### ISSUE TITLE
Prevent dashboard to show context menu

### ISSUE THREAD

**Issue by patrickchartrand** [pruvious/pruvious#66]
Hi 😄 
I would like to thank you for your amazing work with Pruvious, and to ask you one question... Is there a "pruvious way" to deactivate this context menu (Coller/Paste) in the dashboard? 

![image](https://github.com/user-attachments/assets/d0741879-dbd6-4344-83c3-2989541e2532)

This randomly appears on dashbaord and is not very useful for my team. It seems to came from this pruvious composable:

![image](https://github.com/user-attachments/assets/83d103c0-1d19-4f51-8ec5-fcff156d6bb7) 

Again, thank you very much for your work and your time!

**Comment by murisceman** [comment]
Hey, thank you for using Pruvious! I suspect that the context menu may be coming from another source. Does it appear in a different browser? If not, could you provide more details about your application, such as the Nuxt modules you are using?

**Comment by patrickchartrand** [comment]
Yes, it appear on Firefox, Edge and Chrome. It appear only on a specific pattern, ex.: 

1. I'm in my code editor;
2. I go back to my browser;
3. And I click (left or right) anywhere on the dashboard (so when I focus on the dashboard, I suppose).

The pictures above are from Firefox. But here it's from Edge: 
![image](https://github.com/user-attachments/assets/a1eea19a-7a53-4223-8195-eb7f3eca2bc4)

Sorry for this minor issue... 🙈

**Comment by murisceman** [comment]
Ah, I see what's happening now. Thank you for providing more information. Pruvious uses the Clipboard API to manage the copying and pasting of blocks, which triggers this popup in the browser. Unfortunately, there is currently no way to override this composable. I will try to handle this more elegantly in the next version, as it can be quite intimidating for content editors and managers 🫠

**Comment by patrickchartrand** [comment]
I'll stay on the lookout then! Thank you very much 😄

**Comment by brad-sullivan** [comment]
@murisceman is there a reason the data needs to be stored on the clipboard? I just modified it to be stored straight in the `useState(...)` store in the clipboard composable and it seems to work great.

Maybe there's some edge cases I'm missing?

**Comment by murisceman** [comment]
@brad-sullivan The idea was that editors could copy and paste content between two separate sites that share the same block structure. The browser alert appears because Pruvious checks the JSON structure from the clipboard to determine if it can be pasted in different contexts.

**Comment by brad-sullivan** [comment]
@murisceman Ok - thoughts on adding a module option to disable cross-site copying? I can PR.

On Firefox, the little 'Paste' item comes up near constantly and it's really frustrating (I think other browsers handle it better).

I think I found a way to get around using the Clipboard API (so no permissions required) by listening to `paste` events in elements with `contenteditable="true"`, but it would require a bit of work I think. How are you approaching it in v4?

**Comment by murisceman** [comment]
@brad-sullivan If you know how to avoid triggering the browser alert, a PR would be greatly appreciated 🙏

Here is some additional context. These lines are the culprits:

https://github.com/pruvious/pruvious/blob/2c022647298a55a06fc4a3ac92d741965a8db28f/src/runtime/composables/dashboard/clipboard.ts#L30-L45

The clipboard sniffing is used solely for the `canPaste` check in these files:

https://github.com/pruvious/pruvious/blob/2c022647298a55a06fc4a3ac92d741965a8db28f/src/runtime/components/misc/AddBlockPopup.vue#L98-L102

https://github.com/pruvious/pruvious/blob/2c022647298a55a06fc4a3ac92d741965a8db28f/src/runtime/components/misc/MoreBlockOptionsPopup.vue#L154-L159

> How are you approaching it in v4?

I haven't implemented this in v4 yet, and I plan to avoid checking the clipboard contents before any paste event from the user.

**Comment by brad-sullivan** [comment]
@murisceman its not so much to “avoid triggering the browser alert”, but to change the way we interact with the keyboard such that we never need to use the Clipboard API to get on-demand access to the clipboard (which is probably a bit concerning to an educated user anyway - I wouldn’t really want a website to be able to check my clipboard at any time).

The change involves:
- identifying which page elements the user should be allowed to paste onto
- when highlighted and CTRL/CMD is pressed, shift focus to an invisible div with the “contenteditable” attribute.
- listen for the “paste” DOM event on this div. This event will trigger the logic for inserting the block (or whatever will be pasted)

So basically we’re switching from “Pruvious accessing the clipboard” to “the user gives the clipboard data at that moment to pruvious”, this requiring no permissions.

But as you can tell (or maybe I’m wrong) it would require some reworking of the feature to make possible.

My PR would be to add an option to turn off cross-site pasting in the Nuxt config, for those that don’t need it. I’ve actually already implemented that in my fork.

At the moment I don’t have the time to make the PR to fix the issue “properly” (as I described above); as I don’t require cross-site pasting, it’s not a priority for me.

If I get some free time I might be able to take a look, just not in work time.

**Comment by murisceman** [comment]
@brad-sullivan I agree that the clipboard checking functionality should be removed, as it can be intimidating. Instead of using the `contenteditable` container, I would prefer to use [`event.clipboardData`](https://developer.mozilla.org/en-US/docs/Web/API/ClipboardEvent/clipboardData) on `paste` events directly on `window`. Pruvious already has built-in hotkey handling that checks if the user is editing an input field. In that case, we don't want to paste blocks.

https://github.com/pruvious/pruvious/blob/2c022647298a55a06fc4a3ac92d741965a8db28f/src/runtime/composables/dashboard/hotkeys.ts#L105-L108

**Comment by brad-sullivan** [comment]
@murisceman  I haven’t tested on `window` specifically, but I don’t think the paste event will be fired unless focus is on an editable element (hence the contenteditable div)

From the [Mozilla docs on window: paste](https://developer.mozilla.org/en-US/docs/Web/API/Window/paste_event)
> The original target for this event is the [Element](https://developer.mozilla.org/en-US/docs/Web/API/Element) that was the intended target of the paste action

It might not be pretty but other sites seem to handle it (Google sheets, for example)

**Comment by murisceman** [comment]
@brad-sullivan It seems possible with `window`. See here: https://playcode.io/2270820

**Comment by brad-sullivan** [comment]
@murisceman Nice. That simplifies things a lot. I have PR'd this change for your review when you get a moment.

**Comment by Blemming** [comment]
will the fix be merged anytime soon ? the paste context menu is starting to show up on the main website now, not just the admin panel

**Comment by murisceman** [comment]
> [@murisceman](https://github.com/murisceman) I've been away from this for a while so just confirming.. are you waiting on me here?
> 
> Edit: ah.. that's embarrassing. I merged your changes into my fork, and I think my PR to your repo automatically gets that change now.
> 
> Sorry, I'm a bit green to collaborating via GitHub 😅

No worries, thank you for the good work on the PR 💙
